src/hotspot/share/runtime/arguments.cpp

```diff
@@ -1664,7 +1664,7 @@ void Arguments::set_heap_size() {
     reasonable_minimum = limit_heap_by_allocatable_memory(reasonable_minimum);
 
     if (InitialHeapSize == 0) {
-      julong reasonable_initial = (julong)((phys_mem * InitialRAMPercentage) / 100);
+      julong reasonable_initial = (julong)((phys_mem * (julong)InitialRAMPercentage) / 100);
```

Comment:
This and the two changes above loses the fractional part.
```suggestion
      julong reasonable_initial = (julong)((double)phys_mem * InitialRAMPercentage / 100.0);
```